src/main/java/io/cryostat/net/web/http/api/v2/TargetsPostHandler.java

```diff
@@ -129,9 +131,18 @@ public IntermediateResponse<ServiceRef> handle(RequestParameters params) throws
         try {
             MultiMap attrs = params.getFormAttributes();
             String connectUrl = attrs.get("connectUrl");
+
             if (StringUtils.isBlank(connectUrl)) {
                 throw new ApiException(400, "\"connectUrl\" form parameter must be provided");
             }
+            // check incase custom target has short form connection url (i.e `localhost:0`,
+            // etc)
+            if (connectUrl.matches("localhost:\\d+")) {
```

Comment:
This regex pattern should not match only `localhost` - it could be any hostname, or domain name, or IP address here instead.

We'll need to come up with some way to distinguish these "short forms" from fully formed URLs somehow, with a precise definition. I guess what is common to all these options is that they have a host/domain/IP and a port only, and there is no URL scheme/protocol, path, query, etc. So we know the characters `:` and `/` and any whitespace will not appear in the left hand side since these could only be from the scheme or path, and only numbers will appear on the right hand side for the port. Maybe something like this:

```java
connectUrl = connectUrl.strip();
boolean isShortForm = connectUrl.matches("^[^\\s/:]+:\\d+$");
```